TodoList/Tests/TodoList.Services.Tests/Wrappers/DateTimeGeneratorTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using TodoList.Contracts.Services;
+using TodoList.Services.Wrappers;
+
+namespace TodoList.Services.Tests.Wrappers
+{
+    [TestFixture]
+    public class DateTimeGeneratorTest
+    {
+        private IDateTimeGenerator _dateTimeGenerator;
+
+        [SetUp]
+        public void Init()
+        {
+            _dateTimeGenerator = new DateTimeGenerator();
+        }
+
+        [Test]
+        public async Task GetCurrentDateTime_ReturnsDelayedDateTimesCorrectly()
+        {
+            var waitingTime = 100;
```

Comment:
It would be great to make units part of the name, so it is clear what the number actually means. Also, since it is effectively a constant, why not make one of it?

Same applies for _precisionDeviation_. 

Otherwise I like naming here quite a lot 👍